store/v2/db/prefixdb.go

```diff
@@ -55,7 +54,26 @@ func (pdb *PrefixDB) Has(key []byte) (bool, error) {
 	return ok, nil
 }
 
-// Iterator implements RawDB.
+// Set implements corestore.KVStore.
+func (pdb *PrefixDB) Set(key, value []byte) error {
+	if len(key) == 0 {
+		return errors.ErrKeyEmpty
+	}
+	if value == nil {
```

Comment:
personal preference: the key check makes sense but the value check should be ensured in the downstream DB.